### PR TITLE
Task 12 CORE-GET/api/articles-topic-query...

### PR DESCRIPTION
**Pull request by filosoho**
…FEATURE REQUEST the endpoint accepts topic, which filters the articles by the topic value

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -189,6 +201,24 @@ describe("/api/articles", () => {
...
+        .get("/api/articles?topic=non_existent_topic")
+        .expect(404)
+        .then(({ body }) => {
+          expect(body.msg).toBe("404 - Not Found: topic  not found");
+        });
+    });
+
+    test("GET 404: responds with 400 if topic value missing", () => {
+      return request(app)
+        .get("/api/articles?topic=")
```
Comment (MRombi) [inline __tests__/app.test.js:216]: This should be a 200 :)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -189,6 +201,24 @@ describe("/api/articles", () => {
...
+        .expect(400)
+        .then(({ body }) => {
+          expect(body.msg).toBe("400 - Bad Request: Topic value missing");
+        });
+    });
```
Comment (MRombi) [inline __tests__/app.test.js:221]: Let's build a test for when the topic is in the db but has no articles like for the topic paper. We should receive a 200 and an empty array :)

**Review by MRombi — COMMENTED** [review]
Take into account my feedback before proceeding, good job overall :) No need to resubmit

File: __tests__/app.test.js (hunk 1 above)
Comment (filosoho) [reply __tests__/app.test.js:216]: Oh, right! I missed this - thank you very much for pointing this out. :)

File: __tests__/app.test.js (hunk 2 above)
Comment (filosoho) [reply __tests__/app.test.js:221]: Great idea! Thank you again. :)

**Pull request merged**